C/services/south/south.cpp

```diff
@@ -422,11 +422,19 @@ void SouthService::configChange(const string& categoryName, const string& catego
 	// TODO action configuration change
 	logger->info("Configuration change in category %s: %s", categoryName.c_str(),
 			category.c_str());
-	m_configAdvanced = m_mgtClient->getCategory(m_name+"Advanced");
-	try {
-		m_readingsPerSec = (unsigned long)atoi(m_configAdvanced.getValue("readingsPerSec").c_str());
-	} catch (ConfigItemNotFound e) {
-		logger->error("Failed to update poll interval following configuration change");
+	if (categoryName.compare(m_name) == 0)
+	{
+		m_config = ConfigCategory(m_name, category);
+		southPlugin->reconfigure(category);
+	}
+	if (categoryName.compare(m_name+"Advanced") == 0)
+	{
+		m_configAdvanced = ConfigCategory(m_name+"Advanced", category);
+		try {
+			m_readingsPerSec = (unsigned long)atoi(m_configAdvanced.getValue("readingsPerSec").c_str());
```

Comment:
There is a separate issue here that m_readingsPerSec, bufferThreshold and maxSendLatency in advanced config don't take effect until next plugin restart. Later 2 params also need to be read from m_configAdvanced here. Later 2 params need to be passed to Ingest class instance. And readingsPerSec would need re-creating timerFd with correct interval.

Reply:
Created FOGL-2097.